thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AnomaliesResource.java

```diff
@@ -497,6 +485,59 @@ public void createUserAnomaly(@PathParam("anomalyFunctionId") long anomalyFuncti
     mergedAnomalyResultDAO.updateAnomalyFeedback(anomaly);
   }
 
+  /**
+   * Update anomaly feedback for all anomalies of a specific anomaly function within a time range.
+   *
+   * @param startTime start time utc (in millis)
+   * @param endTime end time utc (in millis)
+   * @param functionId anomaly function id
+   * @param feedbackType feedback type
+   */
+  @POST
+  @Path(value = "/updateFeedbackRange/{startTime}/{endTime}/{functionId}")
+  public void updateFeedbackForAnomalyFunctionAndTimeRange(
+      @PathParam("startTime") Long startTime,
+      @PathParam("endTime") Long endTime,
+      @PathParam("functionId") Long functionId,
+      @QueryParam("feedbackType") String feedbackType) {
+
+    if (functionId == null) {
+      throw new IllegalArgumentException("Must provide functionId");
+    }
+
+    if (startTime == null) {
+      throw new IllegalArgumentException("Must provide startTime");
+    }
+
+    if (endTime == null) {
+      throw new IllegalArgumentException("Must provide endTime");
+    }
+
+    if (feedbackType == null) {
+      throw new IllegalArgumentException("Must provide feedbackType");
+    }
+
+    // fetch anomalies
+    List<MergedAnomalyResultDTO> anomalies = mergedAnomalyResultDAO.findByFunctionId(functionId, false);
+
+    // apply feedback
+    for (MergedAnomalyResultDTO anomaly : anomalies) {
+      if (anomaly.getStartTime() < endTime && startTime < anomaly.getEndTime()) {
```

Comment:
Hi Alex, thanks for providing this endpoint. Just realize the updateFeedbackRange should update feedback for the same dimension as with user report anomaly, can we please extend this endpoint to support dimension matching? Thanks

Reply:
pr: https://github.com/linkedin/pinot/pull/2233